src/platform/ESP32/ESP32SecureCertDACProvider.h

```diff
@@ -31,6 +31,27 @@ class ESP32SecureCertDACProvider : public Credentials::DeviceAttestationCredenti
     CHIP_ERROR GetDeviceAttestationCert(MutableByteSpan & outBuffer) override;
     CHIP_ERROR GetProductAttestationIntermediateCert(MutableByteSpan & outBuffer) override;
     CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & messageToSign, MutableByteSpan & outSignBuffer) override;
+
+#ifdef CONFIG_ENABLE_SET_CERT_DECLARATION_API
+    /**
+     * API to set the CD.
+     *
+     * GetCertificationDeclaration() API impl reads the CD from the NVS namespace `chip-factory`.
+     * Use this API to set the CD if it is stored at a different place, eg: embedded in the firmware.
+     * Subsequent reads after calling this API will return the set CD.
```

Comment:
Please document that no copy is made and the underlying data must survive any calls by the stack to reach it.

Reply:
Thanks for the suggestion.